### PR TITLE
Split out-of-stock ammo into its own section

### PR DESCRIPTION
**Pull request by LIYINXUE-PERSONAL**
## Summary
- move out-of-stock ammo out of the main ammo card flow and show it in a dedicated section
- remove zero-round labels from out-of-stock cards and refine the caliber expand/collapse animation
- include the project version bump and add view model test coverage for the new ammo grouping behavior

## Testing
- `xcodebuild test -scheme 'Armory Inventory' -destination 'platform=iOS Simulator,name=iPhone 17,OS=26.4' -only-testing:ArmoryInventoryTests/CaliberListViewModelTests` *(failed in this environment because CoreSimulatorService was unavailable)*

### CODE REVIEW

File: ArmoryInventory/Ammo/CaliberList/CaliberListViewModel.swift [hunk 1]
```diff
@@ -81,6 +81,25 @@ final class CaliberListViewModel {
         }
     }
 
+    func inStockSortedAmmo(for caliber: Caliber) -> [AmmoType] {
+        sortedAmmo(for: caliber).filter { $0.quantity > 0 }
+    }
+
+    func outOfStockSortedAmmo(for caliber: Caliber) -> [AmmoType] {
+        sortedAmmo(for: caliber).filter { $0.quantity == 0 }
```
[the remark this answers is not in the document]
Comment (LIYINXUE-PERSONAL) [reply ArmoryInventory/Ammo/CaliberList/CaliberListViewModel.swift]: Updated `outOfStockSortedAmmo` to classify `<= 0` quantities as out of stock, so negative legacy/corrupted records still show up in the UI and remain editable. I also expanded the view model tests to cover both zero and negative out-of-stock loads.
